packages/central-server/__tests__/sync/CentralSyncManager.startSession.test.js

```diff
@@ -93,6 +94,36 @@ describe('CentralSyncManager.startSession', () => {
     expect(syncSession2).not.toBeUndefined();
   });
 
+  it('A large number of concurrent sessions will not consume the database connection pool', async () => {
+    const centralSyncManager = initializeCentralSyncManager({
+      sync: {
+        lookupTable: {
+          enabled: false,
+        },
+        maxRecordsPerSnapshotChunk: DEFAULT_MAX_RECORDS_PER_SNAPSHOT_CHUNKS,
+        awaitPreparation: true,
+      },
+    });
+    const startSessionPromises = [];
+    for (let i = 0; i < 10; i++) {
+      startSessionPromises.push(centralSyncManager.startSession());
+    }
+
+    const sessionResults = await Promise.allSettled(startSessionPromises);
+
+    const failedSessions = sessionResults.filter(result => result.status === 'rejected');
+    expect(failedSessions.length).toBeGreaterThan(0); // Check to ensure the test is actually testing something
```

Comment:
There's a chance this test might be flaky, as strictly this is a race condition. If so, I'll either bump the number of sessions we try to create, or stub out the SyncSession create to ensure we take all the locks first